src/routes/+layout.svelte

```diff
@@ -1,14 +1,171 @@
-<script>
+<script lang="ts">
+	import MenuItem from '$lib/components/ui/menu/menu-item.svelte'
+	import MenuTitle from '$lib/components/ui/menu/menu-title.svelte'
 	import Typography from '$lib/components/ui/typography.svelte'
+	import Button from '$lib/components/ui/button.svelte'
+	import { Light, SidePanelCloseFilled, SidePanelOpenFilled } from 'carbon-icons-svelte'
 	import '../app.pcss'
+
+	let isMenuOpen = $state(false)
+	let innerWidth: number | undefined = $state()
+
+	const mobileWidth = 700
+
+	function menuOnClick() {
+		// close the menu after a click on mobile
+		if (innerWidth && innerWidth < mobileWidth) {
+			isMenuOpen = false
+		}
+	}
 </script>
 
-<slot />
+<svelte:window bind:innerWidth />
+
+<div class="menu-button-container">
+	<Button variant={isMenuOpen ? 'ghost' : 'overlay'} onclick={() => (isMenuOpen = !isMenuOpen)}>
+		{#if isMenuOpen}
+			<SidePanelCloseFilled size={24} />
+		{:else}
+			<SidePanelOpenFilled size={24} />
+		{/if}
+	</Button>
+</div>
+
+<div class="dark-mode-button-container">
+	<Button variant="overlay"><Light size={24} /></Button>
+</div>
+
+<div class="container">
+	<div class="menu-{isMenuOpen ? 'open' : 'closed'}-placeholder">
+		{#if isMenuOpen}
+			<div class="menu-header"></div>
+			<div class="menu">
+				<MenuTitle content="Diète" bold>
+					<MenuItem href="/" onclick={menuOnClick}>Intro</MenuItem>
+				</MenuTitle>
+				<MenuTitle content="Elements" bold></MenuTitle>
+				<MenuTitle content="Basic components" bold open>
+					<MenuItem href="/components/button" onclick={menuOnClick}>Button</MenuItem>
+				</MenuTitle>
+			</div>
+		{/if}
+	</div>
+	<div class="right">
+		<div class="header"></div>
+
+		<div class="content-container">
+			<div class="content-margin" />
+
+			<div class="content">
+				<slot />
+
+				<section id="footer">
+					<Typography>
+						Made by diete. Source code is available on <a
+							href="https://github.com/diete-design/diete.design">Github</a
+						>.
+					</Typography>
+				</section>
+			</div>
+
+			<div class="content-margin" />
+		</div>
+	</div>
+</div>
 
-<section id="footer">
-	<Typography>
-		Made by diete. Source code is available on <a
-			href="https://github.com/diete-design/diete.design">Github</a
-		>.
-	</Typography>
-</section>
+<style lang="postcss">
+	:global(body) {
+		margin: 0;
+	}
```

Comment:
We can, however at the moment the `app.pcss` contains all the css that is necessary for other components to work properly. I imagine that the workflow could be when one starts to use diete is to copy the content of the `app.pcss` file to their projects along with the component they need.

The `margin` of the body is not required for the components, only for the website so I would not mix them. Maybe we can come up with a solution to extract the necessary css, similarly how we do with the other css files. Or maybe we can also prefix the css variables for the components so that they can be easily distinguished. Nevertheless I feel this needs a longer discussion.

Reply:
Ok I see. I don't quite have an issue with it here then, but in my mind cleaner is to put it in some other global CSS file like `style.pcss` that gets imported in layout. I'll leave it up to you.